src/iceberg_extension.cpp

```diff
@@ -61,18 +62,23 @@ static void SetCatalogSecretParameters(CreateSecretFunction &function) {
 }
 
 static bool SanityCheckGlueWarehouse(string warehouse) {
-	// valid glue catalog warehouse is <account_id>:s3tablescatalog/<bucket>
-	auto end_account_id = warehouse.find_first_of(':');
-	bool account_id_correct = end_account_id == 12;
-	auto bucket_sep = warehouse.find_first_of('/');
-	bool bucket_sep_correct = bucket_sep == 28;
-	if (!account_id_correct) {
-		throw IOException("Invalid Glue Catalog Format: '" + warehouse + "'. Expect 12 digits for account_id.");
-	}
-	if (bucket_sep_correct) {
-		return true;
-	}
-	throw IOException("Invalid Glue Catalog Format: '" + warehouse + "'. Expected '<account_id>:s3tablescatalog/<bucket>");
+    // See: https://docs.aws.amazon.com/glue/latest/dg/connect-glu-iceberg-rest.html#prefix-catalog-path-parameters
+
+	const std::regex patterns[] = {
+        std::regex("^:$"),                   // Default catalog ":" in current account
+        std::regex("^\\d{12}$"),             // Default catalog in a specific account
+        std::regex("^\\d{12}:[^:/]+$"),      // Specific catalog in a specific account
```

Comment:
This one doesn't seem to be listed in the docs ?
If we have `accountID:catalog`, don't we also need `catalog` (`^[^:/]+$`) to access a specific catalog in the current account ?